File: code_examples/4-Functions/1-Understanding_Functions/Java/Main.java
/*
 * This program uses user defined functions
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

final class Main {
  private Main() {
    // Prevent instantiation
    // Optional: throw an exception e.g. AssertionError
    // if this ever *is* called
    throw new IllegalStateException("Cannot be instantiated");
  }

  public static void calculateArea() {
    // input
    Scanner scanner = new Scanner(System.in);
    System.out.print("Enter the length of a rectangle (cm): ");
    int length = scanner.nextInt();
    System.out.print("Enter the width of a rectangle (cm): ");
    int width = scanner.nextInt();

    // process
    int area = length * width;

    // output
    System.out.printf("The area is %d cm²%n%n", area);
  }

  public static void calculatePerimeter() {
    // input
    Scanner scanner = new Scanner(System.in);
    System.out.print("Enter the length of a rectangle (cm): ");
    int length = scanner.nextInt();
    System.out.print("Enter the width of a rectangle (cm): ");
    int width = scanner.nextInt();

    // process
    int perimeter = 2 * (length + width);

    // output
    System.out.printf("The perimeter is %d cm%n%n", perimeter);
  }

  /**
   * Main entry point into program.
   *
   * @param args nothing passed in
   */
  public static void main(final String[] args) {
    // call functions
    calculateArea();
    calculatePerimeter();

    System.out.println("\nDone.");
  }
}

File: code_examples/4-Functions/1-Understanding_Functions/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module uses user defined functions
"""


def calculate_area() -> None:
    """The calculate_area() function calculates area of a rectangle, returns None."""

    # input
    length = int(input("Enter the length of a rectangle (cm): "))
    width = int(input("Enter the width of a rectangle (cm): "))

    # process
    area = length * width

    # output
    print(f"The area is {area} cm²")
    print("")


def calculate_perimeter() -> None:
    """The calculate_perimeter() function calculates perimeter of a rectangle, returns None."""

    # input
    length = int(input("Enter the length of a rectangle (cm): "))
    width = int(input("Enter the width of a rectangle (cm): "))

    # process
    perimeter = 2 * (length + width)

    # output
    print(f"The perimeter is {perimeter} cm")
    print("")


def main() -> None:
    """The main() function just calls other functions, returns None."""

    # call functions
    calculate_area()
    calculate_perimeter()

    print("\nDone.")


if __name__ == "__main__":
    main()
